hive/src/main/java/com/netflix/iceberg/hive/HiveTableOperations.java

```diff
@@ -0,0 +1,235 @@
+    } finally {
+      if (threw) {
+        // if anything went wrong, clean up the uncommitted metadata file
+        deleteFile(newMetadataLocation);
+      }
+      unlock(lockId);
+    }
+
+    refresh();
+  }
+
+  private void setParameters(String newMetadataLocation, Table tbl) {
+    Map<String, String> parameters = tbl.getParameters();
+
+    if (parameters == null) {
+      parameters = new HashMap<>();
+    }
+
+    parameters.put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    parameters.put(METADATA_LOCATION_PROP, newMetadataLocation);
+
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
+      parameters.put(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation());
+    }
+
+    tbl.setParameters(parameters);
+  }
+
+  private StorageDescriptor storageDescriptor(Schema schema) {
+
+    final StorageDescriptor storageDescriptor = new StorageDescriptor();
+    storageDescriptor.setCols(columns(schema));
+    storageDescriptor.setLocation(hiveTableLocation());
+    storageDescriptor.setOutputFormat("org.apache.hadoop.mapred.FileInputFormat");
+    storageDescriptor.setInputFormat("org.apache.hadoop.mapred.FileOutputFormat");
+    SerDeInfo serDeInfo = new SerDeInfo();
+    serDeInfo.setSerdeType(SerdeType.HIVE);
+    serDeInfo.setSerializationLib("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe");
+    storageDescriptor.setSerdeInfo(serDeInfo);
+    return storageDescriptor;
+  }
+
+  private final List<FieldSchema> columns(Schema schema) {
+    return schema.columns().stream().map(col -> new FieldSchema(col.name(), HiveTypeConverter.convert(col.type()), "")).collect(Collectors.toList());
+  }
+
+  private Long acquireLock() throws UnknownHostException, TException {
+    final LockComponent lockComponent = new LockComponent(LockType.EXCLUSIVE, LockLevel.TABLE, database);
+    lockComponent.setTablename(tableName);
+    final LockRequest lockRequest = new LockRequest(Lists.newArrayList(lockComponent),
+            System.getProperty("user.name"),
+            InetAddress.getLocalHost().getHostName());
+    LockResponse lockResponse = metaStoreClient.lock(lockRequest);
+    LockState state = lockResponse.getState();
+    Long lockId = lockResponse.getLockid();
+    //TODO add timeout
+    while (state.equals(LockState.WAITING)) {
+      lockResponse = metaStoreClient.check_lock(new CheckLockRequest(lockResponse.getLockid()));
+      state = lockResponse.getState();
+    }
+
+    if (!state.equals(LockState.ACQUIRED)) {
+      throw new IllegalStateException(String.format("Could not acquire the lock on %s.%s, " +
+              "lock request ended in state %s", database, tableName, state));
+    }
+    return lockId;
+  }
+
+  private void unlock(Optional<Long> lockId) {
+    if (lockId.isPresent()) {
+      try {
+        metaStoreClient.unlock(new UnlockRequest(lockId.get()));
+      } catch (TException e) {
+        throw new RuntimeException(e);
```

Comment:
Should this also have the table name for context? If this fails to unlock, then the table is wedged and we should be able to know what table to fix it, right?

Reply:
Yes we need it, not sure why I missed it. Added it back.